tests/TransactionBuilderTest.php

```diff
@@ -0,0 +1,220 @@
+<?php
+declare(strict_types=1);
+
+use Illuminate\Support\Facades\DB;
+
+class TransactionBuilderTest extends TestCase
+{
+    protected $insertData = ['name' => 'klinson', 'age' => 20, 'title' => 'admin'];
+    protected $originData = ['name' => 'users', 'age' => 20, 'title' => 'user'];
+    protected $connection = 'mongodb_repl';
+    protected $originConnection = 'mongodb';
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        /** change connection to seplset? because the transaction needs  */
+        $this->originConnection = DB::getDefaultConnection();
+        DB::setDefaultConnection($this->connection);
+
+        DB::collection('users')->truncate();
+        DB::collection('users')->insert($this->originData);
+    }
+
+    protected function getEnvironmentSetUp($app)
+    {
+        if (version_compare(env('MONGO_VERSION'), '4', '<')) {
+            $this->markTestSkipped('MongoDB with version below 4 is not supported for transactions');
```

Comment:
Does `MONGO_VERSION` come from the CI environment? If so, I'm not sure what the purpose of this check is as we should be able to assume it's configured correctly. Even within the limited scope of this PR, the check here is inconsistent with the earlier note that transactions require 4.2+ on sharded clusters (granted, those aren't used for CI).

The actual logic for checking whether transactions are supported is a bit more complicated. See [` skipIfTransactionsAreNotSupported()`](https://github.com/mongodb/mongo-php-library/blob/1.13.1/tests/FunctionalTestCase.php#L527) in PHPLIB.

I'm not sure how much of the existing test suite depends on this `MONGO_VERSION` env var, but changing that to a more accurate runtime check might be a worthwhile improvement.

@alcaeus: On a separate note, I wonder if it'd be worthwhile to extract common bits of PHPLIB's test suite as a separate library. I could see it being useful for libraries like this and/or Doctrine ODM.